### DISCUSSION TITLE
How to setup TCP connection like ngrok ?

### DISCUSSION THREAD

**Discussion by airjordan93** [loophole/cli#149] [Q&A]
How do we setup TCP connection like it is available in ngrok?(Which gives domain + port to connect)  I am trying to setup a proxy server from my computer, and ngrok works fine for it, I am wondering how can i do the same in loophole.cloud aswell. Please check the screenshots for Input and excepted output.
![Screenshot_1](https://user-images.githubusercontent.com/95083218/143579871-1989ce95-bb54-4210-843e-4fe9e0ea25db.png)

![Screenshot_2](https://user-images.githubusercontent.com/95083218/143579882-523c411b-383d-49ca-b1ff-29145413a930.png)

**Appreciate any help! Thanks**

**Comment by 0x7f** [comment]
Hi @airjordan93 we do not support TCP connections right now. We do support https tunnels only.

We want to provide a service that is secure by default. All connections are end-to-end encrypted and we are not able to inspect any traffic you host. When providing TCP tunnel, we can not guarantee that the hosted tunnels are encrypted. Among other reasons, this is the most important reason why we do not support TCP tunnels today.
